cloud/main.js

```diff
@@ -33,6 +33,66 @@ Parse.Cloud.define("sendMessage", function(request, response) {
                        });
 });
 
+Parse.Cloud.define("registerNewUser", function(request, response) {
+    getUserCount(function(userCount) {
+        if (userCount == undefined) {
+            response.error();
+            return;
+        }
+
+        var user = new Parse.User();
+        user.set('password', 'password');
+        user.set('username', (userCount + 1).toString());
+
+        user.save(null, {
+            success: function(user) {
+                incrementUserCount(response, function() {
+                    response.success(user);
+                });
+            },
+            error: function(message, error) {
+                response.error(error);
+            }
+        });
+    });
+});
+
+function incrementUserCount(response, successCallback) {
```

Comment:
response should have a success callback already so no need to pass a separate successCallback here